backend/src/show.ts

```diff
@@ -111,3 +116,45 @@ export function getReservedSeats(show_id): Promise<IReservedSeats> {
       return null;
     });
 }
+
+export function create(show: IShow): Promise<IShow> {
+  log.info('ADMIN: Beginning show creation', show);
+  return db.query('insert into nk2_shows (title, venue_id, time, active, inactivate_time, description) values (:title, :venue_id, :time, :active, :inactivate_time, :description)', show)
+    .then((res) => {
+      var show_id = parseInt(res.insertId);
+      log.info('ADMIN: Show created, creating prices', {show_id: show_id});
+      var query_start = 'insert into nk2_prices (show_id, section_id, price, active) values ';
+      var insert_values = _.values(show.sections).map((section: IShowSection) => db.format('(:show_id, :section_id, :price, :active)', _.extend({show_id: show_id}, section)));
+      return db.query(query_start + insert_values.join(','))
+      .then(() => {
+        log.info('ADMIN: Prices created, returning show');
+        return get(show_id, 'backend');
+      });
+    })
+    .catch((err) => {
+      log.error('ADMIN: Creating a show failed', {error: err});
+      return null;
+      throw err;
+    });
+}
+
+export function update(show_id: number, show: IShow): Promise<IShow> {
+  log.info('ADMIN: Beginning show update', show);
+  return db.query('update nk2_shows set title = :title, venue_id = :venue_id, time = :time, active = :active, inactivate_time = :inactivate_time, description = :description where id = :id', show)
+  .then((res) => {
+    log.info('ADMIN: Show updated, updating prices');
+    var query_start = 'insert into nk2_prices (show_id, section_id, price, active) values ';
+    var insert_values = _.values(show.sections).map((section: IShowSection) => db.format('(:show_id, :section_id, :price, :active)', _.extend({show_id: show_id}, section)));
+    var query_end = ' on duplicate key update price = values(price), active = values(active)';
+
+    return db.query(query_start + insert_values.join(',') + query_end).then(() => {
+      log.info('ADMIN: Prices updated, returning show');
+      return get(show_id, 'backend');
+    });
+  })
+  .catch((err) => {
+    log.error('ADMIN: Updating a show failed', {error: err});
+    return null;
```

Comment:
I guess we would like to `throw err`, and `return null` was just for making compiler/linter happy? So, the order of those two is wrong here